lib/context-menu-handlers.js

```diff
@@ -33,16 +32,24 @@ self.on('click', function (node, data) {
   self.postMessage({
     src: node.href || node.src,
     url: window.location.href,
-    domain: host // new URL(node.href || node.src).host
+    domain: host
   });
 });`;
 
 const contextMenuSoundcloudContentScript = `
-self.on('click', function (node) {
-  self.postMessage({
-    url: node.href,
-    domain: 'soundcloud.com'
-  });
+self.on('click', function (node, data) {
+  // if the target element isn't the link we
+  // are looking for, recurse until we find it.
+  function setParentNode(node) {
+    if (node.href) {
```

Comment:
How about `if (node && node.href)`? This would guard against the case where something goes horribly wrong, and you wind up going all the way to the document's `null` parentNode, try to access a property on null, and trigger a TypeError